packages/client/src/lib.js

```diff
@@ -215,6 +217,50 @@ class Web3Storage {
     return res.json()
   }
 
+  /**
+   * @param {Service} service
+   * @param {{before: string, size: number}} opts
+   * @returns {AsyncIterable<Upload>}
+   */
+  static async * listIterator (service, opts) {
```

Comment:
We were careful to keep the API surface area small for the rest of the API so adding 3 methods for this feels like too many.

I'm starting to wonder if `list(options): AsyncIterable` is fine.

First page is just `await client.list().next()` or `for await (const page of client.list())` if you need to paginate through. You can also call `client.list({ before }).next()` with a different date to do your own pagination...

Reply:
yeah, it did feel chonky to have 3 list methods. I think having 1 way to do a thing is better. Folks are going to want a method that gives them an array, but i think if the docs give them the code to copy/paste like

```json
const uploads = []
for await (const item of client.list()) {
  uploads.push(item)
}
```

it should be fine.